sdk/src/main/java/io/dapr/client/domain/HttpExtension.java

```diff
@@ -101,11 +106,31 @@ public HttpExtension(DaprHttp.HttpMethods method) {
     return method;
   }
 
-  public Map<String, String> getQueryString() {
-    return queryString;
+  public Map<String, List<String>> getQueryParams() {
+    return queryParams;
   }
 
   public Map<String, String> getHeaders() {
     return headers;
   }
+
+  /**
+   * Encodes the query string for the HTTP request.
+   * @return Encoded HTTP query string.
+   */
+  public String encodeQueryString() {
+    if ((this.queryParams == null) || (this.queryParams.isEmpty())) {
+      return "";
+    }
+
+    HttpUrl.Builder urlBuilder = new HttpUrl.Builder();
+    // Setting required values but we only need query params in the end.
+    urlBuilder.scheme("http").host("localhost");
```

Comment:
should this be `Properties. SIDECAR_IP` ?

Reply:
See comment right above this line. It is unused, just required by HttpUrl.Builder()